interp/src/primitives.rs

```diff
@@ -27,114 +32,191 @@ pub fn update_cell_state(
 
     match cell_type.id.as_str() {
```

Comment:
We should create an actual struct for the primitives rather than doing string matching. We can then instantiate all the primitive cells once and pass inputs to and from them as needed